### ISSUE TITLE
Can't set a field back to null when insertOnConflictUpdate

### ISSUE THREAD

**Issue by FireLizard** [simolus3/moor#1681]
With the feature [generate_values_in_copy_with](https://drift.simonbinder.eu/docs/advanced-features/builder_options/) one can create a new object with fields set to `null`.

When inserting/updating this object with `InsertStatement.insertOnConflictUpdate` the fields with null are ignored. Its because inside `createContext` the `Insertable.toColumns` is called with `nullToAbsent: true`.

Can we allow null? With `UpdateStatement.replace` it works.

**Comment by simolus3** [comment]
`UpdateStatement.replace` is most commonly used when a row was loaded from the database, modified a bit, and should now be written back. In those cases, we can be pretty sure that fields set to `null` really are meant to be `null`, and not just something that shouldn't change / should fall back to defaults.

When using `insertOnConflictUpdate`, the values is most commonly constructed from Dart, meaning that values generated in the database (e.g. an auto-incrementing key or something that has a default in the schema) are likely set to `null` on the data class. So setting these values back to null doesn't feel like a sane default here.
Also, it would be a bit confusing since we definitely want `nullToAbsent: true` for _regular_  inserts. We could use `nullToAbsent: false` only for the `ON CONFLICT UPDATE` part, but IMO it's still confusing that this would potentially set columns back to `null` when I'm just trying to add a new row.

Long story short: Simple nullability in Dart is not enough to decide between setting back to `NULL` in SQL or using the default / not touching the value. I think companions for inserts are the best solution here, there just isn't enough information for consistent choices otherwise.

Alternatively, you could use this hack to always set `nullToAbsent: false` for specific values:

```dart
extension AlwaysInclude<D> on Insertable<D> {
  Insertable<D> alwaysIncludeColumns() {
    return RawValuesInsertable(toColumns(false));
  }
}
```

Then you'd just have to insert `yourDataClass.alwaysIncludeColumns()` to set your nulls to `NULL` for that particular case.
